src/main/java/net/javadiscord/javabot/api/exception/RestExceptionHandler.java

```diff
@@ -0,0 +1,55 @@
+package net.javadiscord.javabot.api.exception;
+
+import org.jetbrains.annotations.NotNull;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.method.annotation.MethodArgumentTypeMismatchException;
+import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler;
+
+import java.sql.SQLException;
+
+/**
+ * Handles all Rest Exceptions.
+ */
+@ControllerAdvice
+public class RestExceptionHandler extends ResponseEntityExceptionHandler {
+
+	/**
+	 * Handles all {@link MethodArgumentTypeMismatchException}s.
+	 *
+	 * @param e The {@link MethodArgumentTypeMismatchException} which was thrown.
+	 * @return The {@link ResponseEntity} containing the {@link ApiError}.
+	 */
+	@ExceptionHandler({ MethodArgumentTypeMismatchException.class })
+	public ResponseEntity<ApiError> handleMethodArgumentTypeMismatch(@NotNull MethodArgumentTypeMismatchException e) {
+		ApiError error = new ApiError(HttpStatus.BAD_REQUEST, e.getLocalizedMessage(),
+				String.format("%s should be of type %s", e.getName(), e.getRequiredType() != null ? e.getRequiredType().getName() : ""));
+		return new ResponseEntity<>(error, error.status());
```

Comment:
You could create a utility method returning a `ResponseEntity<ApiError>` (always) using `HttpStatus.BAD_REQUEST` with parameters for the exception message and the error(s).